test/Analysis/Inputs/xtu-other.cpp

```diff
@@ -76,6 +76,9 @@ int avtSize(void){
 	return sizeof(avt);
 }
 
+typedef struct { int n; } Anonymous;
```

Comment:
Generally, it is good to have lit tests, but we have the inftastracture to create fine-grained unit tests which do test only the specific case. Could you please create with these code rather a unit test in `ASTImporterTest.cpp`?

Reply:
Unit tests do not work here. There is already a test for this case but it passes without the fix. Maybe it does not handle linkages well? It seems to be a general problem with the unit tests that they are in a too sterile environment and what works in the laboratory may not work in the real environment.

Reply:
I think the unittest we have currently does not test the same.
Perhaps we could try something like:
```
TEST(ImportExpr, ImportTypedefOfUnnamedStruct) {
  MatchVerifier<Decl> Verifier;
  auto Code = "typedef struct { int n; } Anonymous; int declToImport(int n) { Anonymous anon; anon.n = n; return anon.n; }
  EXPECT_TRUE(
      testImport(Code, Lang_CXX, "", Lang_CXX, Verifier, YourMatcherHere()));
}
```

Reply:
I tried it. It does not crash. Probably everything has NoLinkage in the unit tests so no assertion on that.

Reply:
Okay, thanks. I is very strange that we did not crash. Perhaps, - independently from this PR - it is worth to investigate this more why the baseline did not crash with the unit test but crash with the lit test.